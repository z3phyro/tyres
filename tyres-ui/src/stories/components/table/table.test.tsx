import { expect, test, vi } from "vitest";
import { render } from "@solidjs/testing-library";
import Table from "./table";

test("Table renders", () => {
  const { getByRole } = render(() => <Table columns={[]} data={[]} />);

  const tableElement = getByRole("table");
  expect(tableElement).toBeTruthy();
});

test("Table renders with columns", () => {
  const { getByRole } = render(() => (
    <Table
      columns={[{ name: "Test column" }]}
      data={[["Test data"]]}
    />
  ));

  const columnElement = getByRole("columnheader");
  expect(columnElement).toHaveTextContent("Test column");
});

test("Table renders with data", () => {
  const { getByRole } = render(() => (
    <Table
      columns={[{ name: "Test column" }]}
      data={[["Test data"]]}
    />
  ));

  const dataElement = getByRole("cell");
  expect(dataElement).toHaveTextContent("Test data");
});

test("Table renders with actions", () => {
  const { getByRole } = render(() => (
    <Table
      columns={[{ name: "Test column" }]}
      data={[["Test data"]]}
      actions={[{ content: "Test action", action: () => { }, hint: "Test hint" }]}
    />
  ));

  const actionElement = getByRole("button");
  expect(actionElement).toHaveTextContent("Test action");
});

test("Table calls action", () => {
  const action = vi.fn();
  const { getByRole } = render(() => (
    <Table
      columns={[{ name: "Test column" }]}
      data={[["Test data"]]}
      actions={[{ content: "Test action", action, hint: "Test hint" }]}
    />
  ));

  const actionElement = getByRole("button");
  actionElement.click();
  expect(action).toHaveBeenCalled();
});

test("Table renders with custom class", () => {
  const { getByRole } = render(() => (
    <Table
      columns={[{ name: "Test column" }]}
      data={[["Test data"]]}
      class="test-class"
    />
  ));

  const tableElement = getByRole("table");
  expect(tableElement).toHaveClass("test-class");
});

test("Table renders with custom header renderer", () => {
  const { getByRole } = render(() => (
    <Table
      columns={[{ name: "Test column", renderHeader: () => <span>Test header</span> }]}
      data={[["Test data"]]}
    />
  ));

  const headerElement = getByRole("columnheader");
  expect(headerElement).toHaveTextContent("Test header");
});

test("Table renders with custom cell renderer", () => {
  const { getByRole } = render(() => (
    <Table
      columns={[{ name: "Test column", renderCell: () => <span>Test cell</span> }]}
      data={[["Test data"]]}
    />
  ));

  const cellElement = getByRole("cell");
  expect(cellElement).toHaveTextContent("Test cell");
});

test("Table snapshot test", () => {
  const { asFragment } = render(() => (
    <Table
      columns={[{ name: "Test column" }]}
      data={[["Test data"]]}
    />
  ));

  expect(asFragment()).toMatchSnapshot();
});
