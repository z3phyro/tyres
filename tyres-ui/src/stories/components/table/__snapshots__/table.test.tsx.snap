// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Table snapshot test 1`] = `"<section class="w-full overflow-x-auto bg-white p-4 border border-gray-100 rounded overflow-hidden"><table class="w-full min-w-full divide-y divide-gray-300"><thead><tr><th class="pl-4 py-2 uppercase text-gray-700 font-medium text-left text-sm">Test column</th></tr></thead><tbody><tr><td class="pl-4 text-gray-800  text-gray-600 text-lg py-2 pr-4 w-auto">Test data</td></tr></tbody></table></section>"`;
